docs/core/eventdata/tracks.md

```diff
@@ -1,4 +1,730 @@
+for(const auto trackState : track.trackStatesReversed()) {
+  // do something with track state
+}
+
+```
+
+Note that {func}`Acts::TrackProxy::trackStatesReversed` iterates from the {term}`tip state` to
+the {term}`stem state`, i.e. from the outside in.
+
+:::{attention}
+By-default, it is not possible to iterate *forward* through the track states on a track!
+The track's track states need to be *forward-linked* for this to be possible.
+:::
+
+The reason for this is this:
+As the trajectory branches at the second sensor into $S_2$/$S_3$, it is not
+possible to connect the states forward, i.e. store in $S_1$ what the *next*
+state is going to be: it is ambiguous!
+
+However, when track finding has concluded, and the trajectories identified by
+{term}`tip states<tip state>` $S_7$ and $S_8$ have been discarded or are being
+copied into an output container, it is possible to *forward link* the track
+state sequences. This is possible **if** the trajectory does not branch
+anymore! {func}`Acts::TrackProxy::copyFrom` will implicitly forward link the
+track states, as it is guaranteed to not branch after copying.
+
+You can manually add forward-linking to a track by calling
+{func}`Acts::TrackProxy::linkForward` or
+{func}`Acts::TrackProxy::reverseTrackStates`.
+
+:::{warning}
+Calling either {func}`Acts::TrackProxy::linkForward` or
+{func}`Acts::TrackProxy::reverseTrackStates` on a track state sequence which
+has branching will break the branching! If you have other tracks pointing at a
+{term}`tip state` that branches from the sequence you're trying to
+forward-link, it will be corrupted!
+:::
+
+In this example, before any forward linking, the sequence looks like this:
+
+:::{graphviz}
+
+digraph {
+rankdir="LR";
+S2 -> S1;
+S3 -> S1;
+S7 -> S5 -> S4 -> S2;
+S6 -> S3;
+}
+
+:::
+
+After a copy operation of $S_6$ and $S_7$ the resultig track state sequences will look like this:
+
+
+:::{graphviz}
+
+digraph {
+rankdir="LR";
+S11[label="S1 (copy)"];
+S12[label="S1 (copy)"];
+S7 -> S5 -> S4 -> S11;
+S11 -> S4 -> S5 -> S7;
+S6 -> S3 -> S12;
+S12 -> S3 -> S6;
+}
+
+:::
+
+This now includes both forward and backward links, which allows iteration from
+$S_1$/$S_2$ to $S_6$/$S_7$ and the other way around.
+
+Forward iteration can then be achieved like this:
+
+```cpp
+auto track = getTrackFromSomewhere();
+for(const auto trackState : track.trackStates()) {
+  // iterate forward
+  // do something with track state
+}
+```
+
+and the innermost track state becomes directly accessible via
+{func}`Acts::TrackProxy::innermostTrackState`.
+
+## Track State API
+
+:::{doxygenclass} Acts::TrackStateProxy
+:members:
+:::
+
+:::{doxygenclass} Acts::MultiTrajectory
+:members:
+:::
+
+(track_edm_component_sharing)=
+## Component sharing
+
+{class}`Acts::MultiTrajectory` is designed so that components can be shared
+between components. This can be achieved using the
+{func}`Acts::TrackStateProxy::shareFrom` can be used to set this up.
```

Comment:
```suggestion
{class}`Acts::MultiTrajectory` is designed so that components can be shared
between states. This can be achieved using the
{func}`Acts::TrackStateProxy::shareFrom` can be used to set this up.
```
?